http-client-core/src/main/java/io/micronaut/http/client/exceptions/HttpClientException.java

```diff
@@ -50,5 +52,26 @@ public HttpClientException(String message, Throwable cause, boolean shared) {
         if (!shared) {
             throw new IllegalArgumentException("shared must be true");
         }
+        serviceIdLocked = true;
+    }
+
+    public final String getServiceId() {
+        return serviceId;
+    }
+
+    public final void setServiceId(String serviceId) {
+        if (serviceIdLocked) {
+            throw new IllegalStateException("Service ID already set");
+        }
+        this.serviceId = serviceId;
+        serviceIdLocked = true;
+    }
```

Comment:
Needs javadoc with `@since` and `@throws` with an explanation it can only be set once